### PR TITLE
Update docs for writing image comparison tests.

### PR DESCRIPTION
**Pull request by anntzer**
Prefer directing to the function's docstring as the main source of
truth -- testing.rst had some info that was invalid since a while ago,
namely regarding the default value of tol.

Also replace the example image comparison test by a much shorter one.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
My problem is that it's not just the default that became wrong, it's also that we rely on `tol` quite rarely these days anyways and it should definitely not be the first thing on the radar of someone writing a test for the first time.

File: lib/matplotlib/testing/decorators.py [hunk 1]
```diff
@@ -369,9 +369,14 @@ def image_comparison(baseline_images, extensions=None, tol=0,
         only allowed when using pytest.
 
     extensions : [ None | list ]
```
Comment (timhoffm) [inline lib/matplotlib/testing/decorators.py]: While we're at it ``None or list of str`` if you like.

File: doc/devel/testing.rst [hunk 2]
```diff
@@ -152,12 +152,11 @@ tests it::
...
-   def test_hexbin_empty():
-       # From #3886: creating hexbin from empty dataset raises ValueError
-       ax = plt.gca()
-       ax.hexbin([], [])
+   def test_line_dashes():
+       fig, ax = plt.subplots()
+       ax.plot(range(10), linestyle=(0, (3, 3)), lw=5)
 
 The first time this test is run, there will be no baseline image to compare
 against, so the test will fail.  Copy the output images (in this case
```
Comment (anntzer) [inline doc/devel/testing.rst]: @timhoffm the paths below need to be updated as well

Comment (timhoffm) [reply doc/devel/testing.rst]: Done.

**Pull request merged**